### PR TITLE
dotCMS/core#26556 Detect if the AI plugin is installed and enable/disabled the AI Extensions in Block Editor

### LINKED ISSUE #26556: [UI] Not load the AI extensions if the plugin is not properly installed and configured
### Parent Issue
#26495

### User Story
As a user, I want the AI Plugin for the Block Editor to not load if it's not properly installed and configured, ensuring a smooth and error-free experience in the content editing environment.

### Acceptance Criteria
- The AI content and image nodes  should not appear in the Block Editor if  the AI plugin it is not correctly installed and configured.
- The Block Editor should function normally without any disruption or errors due to the AI Plugin not loading.

### dotCMS Version
master

### Proposed Objective
User Experience

### Proposed Priority
Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
- Assumption that the plugin has a mechanism to check its installation and configuration status.
- Need for clear documentation on proper installation and configuration steps for the AI Plugin.

### Quality Assurance Notes & Workarounds
_to_define_

Depend of https://github.com/dotCMS/core/issues/26723

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Detect if the AI plugin is installed and enable/disable the AI Extensions in Block Editor

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:

https://github.com/dotCMS/core/assets/1909643/ce0ed46a-8fa4-42db-954a-495b8ba9b548

### CODE REVIEW

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    get isPluginInstalled(): boolean {
+        return this._isPluginInstalled;
+    }
+
+    /**
+     * Initializes the Block Editor external configurations.
+     *
+     * @return {Promise<any>} A Promise that resolves when the Block Editor is initialized.
+     */
+    initializeBlockEditor(): Promise<unknown> {
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts:25]: Why unknown? Looks like you can know :D

**Review by fmontes — DISMISSED** [review]

File: core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            this._isPluginInstalled = response.status === 200;
+
+            return this._isPluginInstalled;
+        } catch (error) {
+            console.error('Error checking plugin installation:', error);
+
+            this._isPluginInstalled = false;
+
+            return this._isPluginInstalled;
+        }
```
Comment (zJaaal) [inline core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts:55]: This code is kind of duplicated, maybe you can do it once in a `finally` block

**Review by zJaaal — DISMISSED** [review]

File: core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts:55]: sort of, but it depends on the `this._isPluginInstalled = response.status === 200;` I need that assignation after to fill the injectToken—anything different to status 200 needs to be false.  Let me check if I can make it less visually redundant.

File: core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts (hunk 1 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts]: Fixed!

File: core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts]: Improved @zJaaal thanks!..

**Review by fmontes — APPROVED** [review]

**Pull request merged**